playbooks/roce/roce.yaml

```diff
@@ -0,0 +1,143 @@
+---
+# Install Mellanox OFED
+- import_playbook: mofed-install.yaml
+  run_once: true
+  vars:
+    mofed_site_place: "MLNX_OFED-4.6-1.0.1.1"
+    mofed_file_name: "MLNX_OFED_LINUX-4.6-1.0.1.1-ubuntu16.04-x86_64.iso"
+  tags:
+    - mellanox_ofed
+
+# Activate VF for relevant PF
+- hosts: all
+  become: true
+  become_method: sudo
+  tasks:
+    - name: checking the Mellanox OFED is installed
+      stat:
+        path: /usr/bin/ofed_info
+      register: ofed_present
+    - include_tasks: vf-activate.yaml
+      with_items:
+        - "{{ pf1 }}"
+        - "{{ pf2 }}"
+        - "{{ pf3 }}"
+        - "{{ pf4 }}"
+      when: ofed_present.stat.exists
+  tags:
+    - vf_activation
+
+#Install Mellanox GPUDirect - WIP
+
+#Install python dependencies
+- hosts: kube-master

```

Comment:
I would suggest adding the following here:

become: true
become_method: sudo

Without this, the ‘Create configmap’ task in playbooks/roce/k8s-sriov-device-plugin.yaml will fail when the deployment is run by a non-root user (because the ‘openshift’ module will not have been installed within the root user’s Python environment).

Reply:
Because your OS is like 18.04 please use suitable MOFED image in parameter **mofed_file_name** - MLNX_OFED_LINUX-4.6-1.0.1.1-ubuntu18.04-x86_64.iso

Reply:
@vtlrazin:

I used the following mofed vars:
    mofed_site_place: "MLNX_OFED-4.6-1.0.1.1"
    mofed_file_name: "MLNX_OFED_LINUX-4.6-1.0.1.1-ubuntu18.04-x86_64.iso"

I also updated the network definition section to use the vlan_id's that I am using:
        - { network_name: sriov111, res_name: intel.com/sriov_111, vlan_id: 3091 }
        - { network_name: sriov112, res_name: intel.com/sriov_112, vlan_id: 3092 }
        - { network_name: sriov113, res_name: intel.com/sriov_113, vlan_id: 3093 }
        - { network_name: sriov114, res_name: intel.com/sriov_114, vlan_id: 3094 }

I also updated the k8s-cluster.yml playbook with my physical interface device names and dev_id:
    pf1: enp5s0
    pf2: enp12s0
    pf3: enp132s0
    pf4: enp139s0
    dev_id: 1014

I did all of this before running into the issue with the SRIOV Device Plugin pod failing to start up correctly. 

Any other suggestions?

Thanks for your help!!